accounts-db/src/accounts_db/geyser_plugin_utils.rs

```diff
@@ -72,13 +72,18 @@ impl AccountsDb {
         P: Iterator<Item = u64>,
     {
         if let Some(accounts_update_notifier) = &self.accounts_update_notifier {
+            let cached_account = self.accounts_cache.load(slot, pubkey);
```

Comment:
Unfortunately, this approach doesn't do what you hope; at least, it doesn't do it for every transaction. The account will only be present in `accounts_cache` if it has been modified in a previous transaction *in the same slot*. Thus, this will return None a lot of the time, and only return the previous account data for repeated modifications within one slot.

Reply:
Ok let me fix that.

Reply:
Updated the code so that we fetch data from indexes if it is not available in the cache.